backend/src/org/commcare/util/CommCarePlatform.java

```diff
@@ -346,6 +346,29 @@ public void initialize(ResourceTable global) {
         return merged;
     }
 
+    /**
+     * Given an form entry object, return the module's id that contains it.
+     *
+     * @param entry Get the module's id that contains this Entry
+     *
+     * @return The ID of the module that contains the provided entry. Null if
+     * the entry can't be found in the installed suites.
+     */
+    public String getModuleNameForEntry(Entry entry) {
+        Vector<Suite> installed = getInstalledSuites();
+
+        for (Suite s : installed) {
+            for (Entry suiteEntry : s.getEntries().values()) {
+                if (suiteEntry.getCommandId().equals(entry.getCommandId())) {
+                    // XXX PLM: unsure what to do when there are multiple
+                    // entries in getMenus()
+                    return s.getMenus().firstElement().getId();
```

Comment:
So the HQ app builder doesn't really support this, but you can put an <entry> into any number of modules/menus, which is super helpful for certain use cases.